libflux/src/flux/semantic/nodes.rs

```diff
@@ -1578,29 +1588,39 @@ const MINUTES: i64 = SECONDS * 60;
 const HOURS: i64 = MINUTES * 60;
 const DAYS: i64 = HOURS * 24;
 const WEEKS: i64 = DAYS * 7;
-const MONTHS: f64 = WEEKS as f64 * (365.25 / 12.0 / 7.0);
-const YEARS: f64 = MONTHS * 12.0;
-
-// TODO(affo): this is not accurate, a duration value depends on the time in which it is calculated.
-// 1 month is different if now is the 1st of January, or the 1st of February.
-// Some days do not last 24 hours because of light savings.
-pub fn convert_duration(duration: &[ast::Duration]) -> std::result::Result<Duration, String> {
-    let d = duration
-        .iter()
-        .try_fold(0 as i64, |acc, d| match d.unit.as_str() {
-            "y" => Ok(acc + (d.magnitude as f64 * YEARS) as i64),
-            "mo" => Ok(acc + (d.magnitude as f64 * MONTHS) as i64),
-            "w" => Ok(acc + d.magnitude * WEEKS),
-            "d" => Ok(acc + d.magnitude * DAYS),
-            "h" => Ok(acc + d.magnitude * HOURS),
-            "m" => Ok(acc + d.magnitude * MINUTES),
-            "s" => Ok(acc + d.magnitude * SECONDS),
-            "ms" => Ok(acc + d.magnitude * MILLIS),
-            "us" | "µs" => Ok(acc + d.magnitude * MICROS),
-            "ns" => Ok(acc + d.magnitude * NANOS),
-            _ => Err(format!("unrecognized magnitude for duration: {}", d.unit)),
-        })?;
-    Ok(Duration::nanoseconds(d))
+
+// The following durations have month base units
+const MONTHS: i64 = 1;
+const YEARS: i64 = MONTHS * 12;
+
+pub fn convert_duration(ast_dur: &[ast::Duration]) -> std::result::Result<Duration, String> {
+    let negative = ast_dur[0].magnitude.is_negative();
```

Comment:
what if the list is empty or does it always have to be non-empty?

Reply:
I think that the list does have to be non-empty since a duration value is required. Unless it is  possible to have an AST DurationLiteral without a Duration. I can add a check for an empty list here so that we can have a better error message in that case, as well as a test case that handles this.